test/UnitTest/Microsoft.Test.AspNet.OData/TestControllers.cs

```diff
@@ -103,6 +107,35 @@ public TestBadRequestResult(BadRequestResult innerResult)
         }
     }
 
+    /// <summary>
+    /// Wrapper for BadRequestObjectResult
+    /// </summary>
+    public class TestBadRequestObjectResult : TestActionResult
+    {
+        private object _value;
+
+        public TestBadRequestObjectResult(BadRequestObjectResult innerResult)
+            : base(innerResult)
+        {
+            _value = innerResult.Value;
+        }
+
+        public override Task ExecuteResultAsync(ActionContext context)
+        {
+            if (_value is string)
+            {
+                context.HttpContext.Response.StatusCode = 400;
```

Comment:
nit: is there a constant that we can use from anywhere?

Reply:
Yes, in ASP.NET Classic, but not in ASP.NET Core.